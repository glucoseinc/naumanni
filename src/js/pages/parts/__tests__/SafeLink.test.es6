import React from 'react'
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'

import {SafeLink} from '../SafeParts'


const takeSnapshot = (component) => {
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
}

describe('SafeLink', () => {
  let component

  describe('http link', () => {
    beforeAll(() => {
      component = <SafeLink href="http://example.com" />
    })
    it('allowed as href', () => {
      const wrapper = shallow(component)
      expect(wrapper.prop('href')).toBe('http://example.com')
      takeSnapshot(component)
    })
  })

  describe('https link', () => {
    beforeAll(() => {
      component = <SafeLink href="https://example.com" />
    })
    it('allowed as href', () => {
      const wrapper = shallow(component)
      expect(wrapper.prop('href')).toBe('https://example.com')
      takeSnapshot(component)
    })
  })

  describe('javascript link', () => {
    beforeAll(() => {
      component = <SafeLink href={'javascript:alert(document.cookie)'} />
    })
    it('will be converted to "javascript:void(0)"', () => {
      const wrapper = shallow(component)
      expect(wrapper.prop('href')).toBe('javascript:void(0)')
      takeSnapshot(component)
    })
  })

  describe('children', () => {
    beforeAll(() => {
      component = (
        <SafeLink href="http://example.com">
          {'spam'}
        </SafeLink>
      )
    })
    it('will be passed as its children', () => {
      const wrapper = shallow(component)
      expect(wrapper.text()).toBe('spam')
      takeSnapshot(component)
    })
  })

  describe('props', () => {
    beforeAll(() => {
      component = <SafeLink href="http://example.com" target="_blank" />
    })
    it('will be passed as attributes', () => {
      const wrapper = shallow(component)
      expect(wrapper.prop('target')).toBe('_blank')
      takeSnapshot(component)
    })
  })
})
