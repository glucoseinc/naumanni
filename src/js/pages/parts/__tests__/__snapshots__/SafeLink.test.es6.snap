// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SafeLink children will be passed as its children 1`] = `
<a
  href="http://example.com"
>
  spam
</a>
`;

exports[`SafeLink http link allowed as href 1`] = `
<a
  href="http://example.com"
/>
`;

exports[`SafeLink https link allowed as href 1`] = `
<a
  href="https://example.com"
/>
`;

exports[`SafeLink javascript link will be converted to "javascript:void(0)" 1`] = `
<a
  href="javascript:void(0)"
/>
`;

exports[`SafeLink props will be passed as attributes 1`] = `
<a
  href="http://example.com"
  target="_blank"
/>
`;
